saleor/payment/gateways/np_atobarai/__init__.py

```diff
@@ -0,0 +1,156 @@
+import logging
+import os
+from typing import List
+
+from ....order.models import Fulfillment
+from ... import PaymentError, TransactionKind
+from ...interface import GatewayConfig, GatewayResponse, PaymentData
+from ...models import Payment
+from . import api
+from .api_types import ApiConfig, PaymentStatus, get_api_config
+from .utils import (
+    get_fulfillment_for_order,
+    get_payment_name,
+    get_shipping_company_code,
+    notify_dashboard,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def inject_api_config(fun):
+    def inner(payment_information: PaymentData, config: GatewayConfig):
+        return fun(payment_information, get_api_config(config.connection_params))
+
+    return inner
+
+
+def parse_errors(errors: List[str]) -> str:
+    # FIXME: better solution?
+    #  Transaction.error in database has max_length of 256
+    return os.linesep.join(errors)[:256]
+
+
+@inject_api_config
```

Comment:
Do we really need that decorator? Why not just call `get_api_config` in function body?